### PR TITLE
Feat/join groups

### PR DESCRIPTION
**Pull request by Lil-Noe**
This pull request includes multiple changes to improve the organization and functionality of the groups, including a new screen called JoinGroupScreen which allows to search for an existing group or create a new one.

Addition of new group-related files:
* Added a new screen in `src/app/views/groups/join_group_screen.tsx`.
* Added the corresponding ViewModel in `src/app/viewmodels/groups/joinGroupScreenViewModel.tsx`.
* Added test cases for the `JoinGroupScreen` component in `__tests__/View-tests/groups/joinGroupViewModel.test.tsx`.

Addition of new test cases:
* Added a comprehensive test suite for the `GroupListItem` component in `__tests__/View-tests/components/groups/groupListItem.test.tsx`.
* Added detailed test cases for the `ListOfFilteredGroups` component in `__tests__/View-tests/components/groups/listOfFilteredGroups.test.tsx`.
* Added test cases for the `JoinGroupScreen` component in `__tests__/View-tests/groups/joinGroupScreen.test.tsx`.

### Tooling and Configuration:

* [`.vscode/extensions.json`](diffhunk://#diff-c16655a98a3ee89a7636a59c59a72b0e93649e3a1e947327cfc43a1336b4e912R1-R3): Added a recommendation for the `orta.vscode-jest` extension to enhance the development environment for testing.

### CODE REVIEW

**Review by margdsq — CHANGES_REQUESTED** [review]
Great PR Lilian ! I love this feature and the design you implemented. I have a few things to ask for, mostly tests for edge case, and insuring we still have a good test coverage, but it was great ! 
Good job🎉

File: __tests__/View-tests/components/groups/groupListItem.test.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        handleJoin={mockHandleJoin}
+      />,
+    );
+
+    expect(getByText("JOIN")).toBeTruthy();
+    const joinButton = getByTestId("join-button-TestGroup");
+
+    await waitFor(() => {
+      fireEvent.press(joinButton);
+    });
```
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply __tests__/View-tests/components/groups/groupListItem.test.tsx:82]: It seems it's even faster thank you

File: __tests__/View-tests/components/groups/groupListItem.test.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import { fireEvent, render, waitFor } from "@testing-library/react-native";
+import { GroupListItem } from "@/src/views/components/groups/group_list_item";
+
+/**
+ * This test suite is for the GroupListItem component
+ * UI-part of the GroupListItem component is tested
+ */
+describe("GroupListItem Component", () => {
+  const mockHandleJoin = jest.fn();
+
```
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply __tests__/View-tests/components/groups/groupListItem.test.tsx:11]: Thanks for reminding be this good practice !

File: __tests__/View-tests/groups/createGroupScreen.test.tsx
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply __tests__/View-tests/groups/createGroupScreen.test.tsx:1]: No it seems it's not possible to do an event with Jest that would slide it and we can't even mock it.

File: __tests__/ViewModel-tests/group/joinGroupViewModel.test.tsx
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply __tests__/ViewModel-tests/group/joinGroupViewModel.test.tsx:1]: Good idea thank you (already have done the empty search text)

File: __tests__/ViewModel-tests/group/joinGroupViewModel.test.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+describe("useJoinGroupViewModel", () => {
+  const mockFirestoreCtrl = new FirestoreCtrl();
+  const uid = "tester-uid";
+
+  // Before each test, mock the console info and clear all mocks
+  beforeEach(() => {
+    jest.spyOn(console, "info").mockImplementation(() => {});
+    jest.clearAllMocks();
+  });
+
```
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply __tests__/ViewModel-tests/group/joinGroupViewModel.test.tsx:44]: I already do but in different test cases.

File: app/src/viewmodels/camera/CameraViewModel.tsx [hunk 4]
```diff
@@ -79,15 +79,13 @@ export default function useCameraViewModel(
 
   // Take a picture with the camera
   const takePicture = async () => {
-    if (camera.current) {
```
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply app/src/viewmodels/camera/CameraViewModel.tsx:82]: Thank you !

File: app/src/views/components/groups/group_list_item.tsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+          text={name}
+          textStyle={styles.titleIcon}
+          textColorType="textOverLight"
+          testID="group-pressable-button"
+        ></ThemedTextButton>
+      </ThemedView>
+
+      {/* Display the group's challenge */}
+      <ThemedView style={styles.textContainer}>
+        <ThemedText style={styles.groupChallenge}>{challengeTitle}</ThemedText>
```
[the remark this answers is not in the document]
Comment (Lil-Noe) [reply app/src/views/components/groups/group_list_item.tsx:41]: That's true ! But I think it would be more intuitive to display an empty string instead of another message saying there is no challenge title.

File: __tests__/View-tests/groups/createGroupScreen.test.tsx
[the remark this answers is not in the document]
Comment (margdsq) [reply __tests__/View-tests/groups/createGroupScreen.test.tsx:1]: Oh too bad !

File: app/src/views/components/groups/group_list_item.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (margdsq) [reply app/src/views/components/groups/group_list_item.tsx]: It should never happens anyway but ok !
